lib/std/jsfetch.nim

```diff
@@ -0,0 +1,203 @@
+    ## `fetch()` API, simple `GET` only, returns a `Future[Response]`.
+
+  proc fetch*(url: cstring or Request, options: FetchOptions): Future[Response] {.importjs: "$1(#, #)".}
+    ## `fetch()` API that takes a `FetchOptions`, returns a `Future[Response]`.
+
+  func toCstring*(self: Request or Response or Body or FetchOptions): cstring {.importjs: "JSON.stringify(#)".}
+
+  func `$`*(self: Request or Response or Body or FetchOptions): string = $toCstring(self)
+
+
+runnableExamples("-d:nimExperimentalJsfetch"):
+  import std/[httpcore, asyncjs, sugar, jsconsole, jsheaders, jsformdata]
+
+  if defined(nimJsFetchTests):
+    block:
+      let options0: FetchOptions = unsafeNewFetchOptions(
+        metod = "POST".cstring,
+        body = """{"key": "value"}""".cstring,
+        mode = "no-cors".cstring,
+        credentials = "omit".cstring,
+        cache = "no-cache".cstring,
+        referrerPolicy = "no-referrer".cstring,
+        keepalive = false,
+        redirect = "follow".cstring,
+        referrer = "client".cstring,
+        integrity = "".cstring
+      )
+      doAssert options0.keepalive == false
+      doAssert options0.metod == "POST".cstring
+      doAssert options0.body == """{"key": "value"}""".cstring
+      doAssert options0.mode == "no-cors".cstring
+      doAssert options0.credentials == "omit".cstring
+      doAssert options0.cache == "no-cache".cstring
+      doAssert options0.referrerPolicy == "no-referrer".cstring
+      doAssert options0.redirect == "follow".cstring
+      doAssert options0.referrer == "client".cstring
+      doAssert options0.integrity == "".cstring
+      doAssert options0.toCstring is cstring
+
+    block:
+      let options1: FetchOptions = newFetchOptions(
+        metod =  HttpPost,
+        body = """{"key": "value"}""".cstring,
+        mode = fmNoCors,
+        credentials = fcOmit,
+        cache = fchNoCache,
+        referrerPolicy = frpNoReferrer,
+        keepalive = false,
+        redirect = frFollow,
+        referrer = "client".cstring,
+        integrity = "".cstring
+      )
+      doAssert options1.keepalive == false
+      doAssert options1.metod == $HttpPost
+      doAssert options1.body == """{"key": "value"}""".cstring
+      doAssert options1.mode == $fmNoCors
+      doAssert options1.credentials == $fcOmit
+      doAssert options1.cache == $fchNoCache
+      doAssert options1.referrerPolicy == $frpNoReferrer
+      doAssert options1.redirect == $frFollow
+      doAssert options1.referrer == "client".cstring
+      doAssert options1.integrity == "".cstring
+      doAssert options1.toCstring is cstring
+
+    block:
+      let response: Response = newResponse(body = "-. .. --".cstring)
+      doAssert response.clone() is Response ## Cloned.
+      doAssert response.toCstring is cstring
+      let request: Request = newRequest(url = "http://nim-lang.org".cstring)
+      doAssert request.clone() is Request   ## Cloned.
```

Comment:
* this test fails at RT in browser
* avoid such static tests (eg `a is Foo`) at RT since we might as well run them at CT:
=> `static: assert request.clone() is Response`